### ISSUE TITLE
Default provider configuration required even when not needed

### ISSUE THREAD

**Issue by lukehoban** [pulumi/pulumi-cloudflare#2]
From a user:

I’m trying to use the Cloudflare provider, and would like to configure the provider instance myself and passing it to the resources with the opts.provider field. However just having the following code

```typescript
       const cfProvider = new cloudflare.Provider(“CF-Provider”,
           {
               email: “no@no.to”,
               token: “abrakadabra”
           }
       );
```

causes the following error when doing “pulumi up” (there are no Cloudflare resources in my code yet):

```
   error: Missing required configuration variable ‘cloudflare:email’
       please set a value using the command `pulumi config set cloudflare:email <value>`
   error: an unhandled error occurred: Program exited with non-zero exit code: 1
```

**Comment by lukehoban** [comment]
Note that this may be related to https://github.com/pulumi/pulumi-terraform/issues/292.

**Comment by jen20** [comment]
Fixed by #11 - I'll tag a release as soon as CI is finished.

**Comment by jen20** [comment]
Fixed in v0.17.3.
